File: day11_string_manipulation/permutation.cpp
#include<iostream>
#include<string>
#include<map>
#include <cstring> 
#include <cctype> 

using namespace std ;


bool checkPermutation(string str) {
    map<char,int> charMap;
    for (int i = 0; i < str.length(); i++)
    {
        // if (str[i] != ' ')
        if (isspace(str[i])!=true)
        {
            charMap[str[i]]++;
        }
        
    }
    int oddCount= 0;
    bool isPermutation= true;

    for (int i = 0; i < str.length(); i++)
    {
        if (str[i]%2==1)
        {
            oddCount++;
        }
        if (oddCount >1)
        {
            isPermutation = false;
            break;
        }
        
        
    }
    return isPermutation;    
}

int main(int argc, char const *argv[])
{
    cout<<checkPermutation("Taaut")<<endl;
    cout<<checkPermutation("Tact Coa");
    return 0;
}

File: day11_string_manipulation/permutation.go
package main

import "fmt"

func permutationPalindrome(str string) bool {
	charMap := make(map[rune]int)
	for _, char := range str {
		if char != rune(' ') {
			charMap[char]++
		}
	}
	oddCount := 0
	isPermutation := true

	for _, value := range charMap {
		if value%2 == 1 {
			oddCount++
		}
		if oddCount > 1 {
			isPermutation = false
			break
		}
	}
	return isPermutation

}

func main() {
	fmt.Println(permutationPalindrome("Taaut"))
	fmt.Println(permutationPalindrome("Tact Coa"))
}
